File: fib/src/fib_serial.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int fib(int n)
{
  int x,y;

  if(n<2) return n;
  x= fib(n-1);
  y= fib(n-2);
  return x+y;
}

int main (int argc, char *argv[])
{
  int N;
  int res;
  clock_t t_start,t_end;
  
  N=44;

  t_start=clock();
  res=fib(N);
  t_end=clock();
  printf("res=%d\n",res);
  printf("Took %g s\n",(double)(t_end - t_start)/CLOCKS_PER_SEC);

  return EXIT_SUCCESS;
}

File: fib/src/fib_serial.f90
PROGRAM fib_serial
  IMPLICIT NONE
  INTEGER :: fib
  INTEGER :: N, res
  REAL :: t_start, t_end
  N=44
  CALL cpu_time(t_start)
  res=fib(N)
  CALL cpu_time(t_end)

  WRITE(*,*) "result = ",res
  WRITE(*,*) "Expended wall clock time = ",t_end-t_start
  
END PROGRAM fib_serial
  
RECURSIVE FUNCTION fib(n) RESULT(res)
  IMPLICIT NONE

  INTEGER, INTENT(IN) :: n
  INTEGER :: res

  INTEGER :: x,y
  
  IF (n<2) THEN
     res=n
  ELSE
     x = fib(n-1)
     y = fib(n-2)
     res=x+y
  END IF
END FUNCTION fib
